content/en/blog/2020/large-scale-security-policy-performance-tests/index.md

```diff
@@ -0,0 +1,158 @@
+subtitle: The effect of security policies on latency of requests
+description: The effect of security policies on latency of requests.
+publishdate: 2020-09-10
+attribution: "Michael Eizaguirre (Google), Yangmin Zhu (Google), Carolyn Hu (Google)"
+keywords: [test,security policy,performance]
+---
+
+## Overview
+
+Istio has a wide range of Security policies which can be easily configured into systems of services. As the number of applied Security policies increases, it is important to understand the relationship of latency, memory usage, and CPU usage of the system.
+
+This blog post goes over common Security policies use cases and how the number of Security policies or the number of specific rules in a Security policy can affect the overall latency of requests.
+
+## Setup
+
+There are a wide range of Security policies and many more combinations of those policies. We will go over 6 of the most commonly used test cases.
+
+The following test cases are run in an environment which consists of a Fortio client sending requests to a Fortio server, with a baseline of no Envoy sidecars deployed. The following data was gathered by using the performance benchmarking tool that can be found [here](https://github.com/istio/tools/tree/master/perf/benchmark).
+{{< image width="55%" ratio="45.34%"
+    link="istio_setup.svg"
+    alt="Environment setup"
+    >}}
+
+In these test cases, requests either do not match any rules or match only the very last rule in the security policies. This ensures that the RBAC filter is applied to all policy rules, and never matches a policy rule before before viewing all the policies. Even though this is not necessarily what will happen in your own system, this policy setup provides data for the worst possible performance of each test case.
+
+## Test cases
+
+1. MTLS STRICT vs plaintext.
+
+1. A single authorization policy with a variable number of principal rules as well as a `PeerAuthentication` policy. The principal rule is dependent on the`PeerAuthentication` policy being applied to the system.
+
+1. A single authorization policy with a variable number of `requestPrincipal` rules as well as a `RequestAuthentication` policy. The `requestPrincipal` is dependent on the `RequestAuthentication` policy being applied to the system.
+
+1. A single authorization policy with a variable number of paths vs `sourceIP` rules.
+
+1. A variable number of authorization policies consisting of a single path or `sourceIP` rule.
+
+1. A single `RequestAuthentication` policy with variable number of `JWTRules` rules.
+
+## Data
+
+The y-axis of each test is the latency in milliseconds, and the x-axis is the number of concurrent connections. The x-axis of each graph consists of 3 data points that represent a small load (qps=100, conn=8), medium load (qps=500, conn=32), and large load (qps=1000, conn=64).
+
+{{< tabset category-name="platform" >}}
+
+{{< tab name="MTLS vs plainText" category-value="one" >}}
```

Comment:
tab namings are confusing.  The first tab said mtls vs plain text, however, you seem to have plain text for each other tabs as well (for the baseline without sidecar).

also for your sidecar with no policy, i assume default is permissive, or strict?

Reply:
So in this graph we wanted to compare how mtls strict compares to the baseline, in other tabs the baseline was also in the graph so that we can have a consistent comparison through all graphs. With no policies the default should be strict